### ISSUE TITLE
Accessibility regression: "Save draft", "Switch to draft", and "Preview" buttons can't be visually identified as user interface controls

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#23890]
**Describe the bug**
In the WordPress 5.5 block editor, the "Save draft", "Switch to draft", and "Preview" buttons have been restyled and in their normal state they now look like plain text.

For accessibility, and I'd say also for a good UI, they should be styled in a way that they can be identified as user interface controls at first glance.

In WordPress 5.4, they weren't perfect either but at least:
- the "Save draft" button, although it missed a shape or an underline, had blue text: blue is a well established convention in WordPress for interactive controls
- same for the "Switch to draft" button
- the "Preview" button looked like a button, which was OK

Screenshots:

<img width="412" alt="save draft preview buttons 5 4" src="https://user-images.githubusercontent.com/1682452/87246990-42c7b200-c451-11ea-9bd0-6c792f921ec4.png">

<img width="427" alt="Screenshot 2020-07-12 at 15 02 58" src="https://user-images.githubusercontent.com/1682452/87246995-48bd9300-c451-11ea-920c-66c06f37ad8c.png">

Instead, now in WordPress 5.5 they look like plain text:
- in their normal state, there's no shape or underline to identify them as buttons
- the buttons text color is a dark grey `#1e1e1e` which makes them look as plain text

Screenshots:

<img width="410" alt="save draft preview buttons 5 5" src="https://user-images.githubusercontent.com/1682452/87247027-74d91400-c451-11ea-9b6e-2992af204490.png">

<img width="412" alt="Screenshot 2020-07-12 at 15 02 33" src="https://user-images.githubusercontent.com/1682452/87247029-799dc800-c451-11ea-87e3-4bba3983ae7e.png">

Recommended:
- either restore the blue color and add an underline
- or, preferably, make them look like buttons, because that's what they are

**Comment by afercia** [comment]
As far as I can tell, these buttons style was changed in #21192.

**Comment by MichaelArestad** [comment]
Have you heard any reports of confusion around this change?

These buttons are consistent with the rest of the actions in the toolbar. The buttons have very visible focus/hover states that clearly show them as a button. Almost every item in the toolbar is a button or action. Anything that is disabled is greyed out. The "Saved" text is also greyed out.

**Comment by afercia** [comment]
I do realize now this is an issue with _all_ the new buttons styling.

> The buttons have very visible focus/hover states that clearly show them as a button

The point is in the _normal_ state, these new buttons don't look like buttons. They look like text. And they're all over the interface now.

**Comment by MichaelArestad** [comment]
> The point is in the normal state, these new buttons don't look like buttons. They look like text. And they're all over the interface now.

I understand what you're saying. I have not seen any evidence that it is an issue in the toolbars. In both Mac and Windows operating systems, there is a window toolbar where "File," "Edit," and "Help" items live that don't have any distinguishing button characteristics yet function as buttons. You can see similar patterns in many Google applications (Docs, etc). This works because there isn't a mix of text and buttons. They are all buttons/actions. Both the block toolbar and the top toolbar only have actions. There is no text mixed in.

If there are specific cases where buttons are next to text, we should visually differentiate them and we can do so using the other button styles (secondary, etc). For the toolbars, my recommendation is for them to remain as they are currently.

**Comment by enriquesanchez** [comment]
I agree that this is not an issue because by being in the editor's toolbar, these actions are visually evident as such from page design and context.

**Comment by afercia** [comment]
Thanks @MichaelArestad I do understand your point. In a way, someone may consider this issue similar to the recommendation for _underlined links_ where, when links can be identified as such by the context, they don't necessarily need to be underlined.

However, I think this is a bit different. For low vision color impairment, etc, this really looks just as text.

> This works because there isn't a mix of text and buttons. 

Not entirely correct: in the toolbar, the "Saved" text is just text and vor users with vision problems, it looks exactly like "Preview". 

More importantly, I see this is now the styling for all the default buttons all across the admin. For example in the block placeholders:

<img width="659" alt="Screenshot 2020-07-23 at 17 08 12" src="https://user-images.githubusercontent.com/1682452/88303694-9cf52c80-cd07-11ea-8077-b5c266184a43.png">

To me, this is an issue. I'd bring it to the attention of the accessibility team in the next weekly meeting to see what others think.

**Comment by afercia** [comment]
Noting that also controls like "Move to trash" look now just as red text:

<img width="309" alt="Screenshot 2020-07-23 at 18 07 02" src="https://user-images.githubusercontent.com/1682452/88310023-7cc96b80-cd0f-11ea-8fe3-c0115b2383d3.png">

While in the Classic editor, "Move to trash" is underlined and for good reasons, even though it's close to other interactive controls:

<img width="323" alt="Screenshot 2020-07-23 at 18 07 21" src="https://user-images.githubusercontent.com/1682452/88310175-a4203880-cd0f-11ea-8662-f5920976e2ac.png">

**Comment by MichaelArestad** [comment]
> Not entirely correct: in the toolbar, the "Saved" text is just text and vor users with vision problems, it looks exactly like "Preview".

Right. I mentioned that above: "Anything that is disabled is greyed out. The "Saved" text is also greyed out." The copy, styling and context distinguishes it from the rest of the actions in the toolbar.

We could change the top toolbar buttons "Preview" and "Save draft" to use the blue text styling. It will bring the styling in line with the buttons in the block placeholders.

**Comment by shaunandrews** [comment]
Its worth noting that we have plenty of other examples where buttons with only text are used in similar manners to indicate an action:

<img width="333" alt="image" src="https://user-images.githubusercontent.com/191598/88426486-43503900-cdbf-11ea-8c1a-571624589b3a.png">

<img width="256" alt="image" src="https://user-images.githubusercontent.com/191598/88426435-2d427880-cdbf-11ea-8b58-14ffc6cf832f.png">

<img width="231" alt="image" src="https://user-images.githubusercontent.com/191598/88426502-4e0ace00-cdbf-11ea-8674-d12dffa98480.png">

<img width="164" alt="image" src="https://user-images.githubusercontent.com/191598/88426515-52cf8200-cdbf-11ea-84cf-dd9dd9857558.png">

<img width="153" alt="image" src="https://user-images.githubusercontent.com/191598/88426559-62e76180-cdbf-11ea-8192-72700371dcd6.png">

<img width="245" alt="image" src="https://user-images.githubusercontent.com/191598/88426605-71357d80-cdbf-11ea-9160-1417dd97bdd1.png">

This is just in the block editor. There's lot of examples throughout the rest of WordPress, too.

**Comment by enriquesanchez** [comment]
I come back to my previous comment about these being identified as interactive elements based on context and page design. They exist in the context of a toolbar, as such they inherit meaning.

A few examples where this meaning by context situation also occurs:

**Google Docs**

<img width="559" alt="Screen Shot 2020-07-28 at 15 28 18" src="https://user-images.githubusercontent.com/3276087/88718555-af94aa80-d0e7-11ea-92b4-3a0478c78a77.png">

File, Edit, View, Insert, etc. They're all interactive elements and are understood as such based on context and page design.

**macOS**

<img width="585" alt="Screen Shot 2020-07-28 at 15 29 35" src="https://user-images.githubusercontent.com/3276087/88718768-dfdc4900-d0e7-11ea-86df-3ea3d6d6242b.png">

File, Edit, View, History, etc. Same as above, they don't need to look like buttons or links because by context they're inheriting the interactive meaning.

**W3C**

<img width="505" alt="Screen Shot 2020-07-28 at 15 29 54" src="https://user-images.githubusercontent.com/3276087/88718967-2af65c00-d0e8-11ea-9562-b84f022e2aa5.png">

The items in the table of contents don't have an underline or different color. They are understood as links by way of the context they're in. 

---

The above is also happening with Gutenberg's toolbar. _It is_ a toolbar and as such, it is understood that the items inside it are interactive elements.

Now, all of the examples above are not doing that because they want a "minimalist" or "clean" user interface. This is done on purpose for balance and hierarchy of information. It aids with cognitive load and helps users understand and find they way around the interface.

**Comment by afercia** [comment]
@MichaelArestad 

> The "Saved" text is also greyed out." The copy, styling and context distinguishes it from the rest of the actions in the toolbar.

Then I'm afraid we agree we disagree 🙂  To me, this (see screenshot below) is _not_ sufficiently distinguishable:

<img width="189" alt="Screenshot 2020-08-06 at 16 36 17" src="https://user-images.githubusercontent.com/1682452/89545239-64734980-d803-11ea-8034-5c1b6afe5bad.png">

@shaunandrews 

> Its worth noting that we have plenty of other examples where buttons with only text are used in similar manners to indicate an action:

The examples in your screenshots are from menus. In a menu, I'd expect all the items are interactive buttons. This is a case where I'd agree context clarifies what these text are.

@enriquesanchez 

Your first two examples are from operating systems UIs. That's a very different case. The last example is a navigation menu made of links. As you may know, the WordPress accessibility coding standards themselves state that in that case links don't necessarily need to be underlined: https://developer.wordpress.org/coding-standards/wordpress-coding-standards/accessibility/#links-underline-or-no-underline

I'd also like to know if this change to the buttons has ever been user tested and whether there are data that show it benefits _everyone_. I do realize that would broaden the discussion a bit too much, though.

Whatever the design arguments are, one thing is pretty clear to me. In WordPress 5.4 the controls highlighted in the screenshot above were clearly identifiable as interactive controls (with the exception of "Save draft"): 

<img width="1281" alt="Screenshot 2020-08-06 at 16 45 07" src="https://user-images.githubusercontent.com/1682452/89546996-81108100-d805-11ea-93cf-2e9f73ebf3b0.png">

In WordPress 5.5, they aren't:

<img width="1283" alt="Screenshot 2020-08-06 at 16 46 01" src="https://user-images.githubusercontent.com/1682452/89547036-8e2d7000-d805-11ea-899b-4f069673867e.png">

To any web accessibility specialist, this looks like an obvious regression in the accessibility level compared to the previous version.

Latest conversation on Slack during the accessibility bug scrub special edition last Tuesday: https://wordpress.slack.com/archives/C02RP4X03/p1596554991310600

Most relevant points:
- Some first (admittedly anecdotal) feedback from users, pretty confused by this change.
- The only fact there’s a large disagreement on this change it’s a good reason to revert and better consider in the next cycle.
- We (accessibility team) all realize this change is part of an overall design shift and handling this more holistically would be important; but that can't happen in 5.5.
- However, what can be done now for 5.5? Not sure a revert, per se, is the best option but restoring a design that makes it clear that these are controls would help.
- from @joedolson: If we make a request, I think it needs to be very specific; preferable would be restoring the design characteristics previously in place, and address this more systematically for 5.6.
- from @afercia: I’d second @joedolson as it seems the safest option to me, for now.

In absence of a solution, if WordPress 5.5 is released with the current buttons design, I'm afraid the accessibility team won't have any option other than considering this an obvious regression in the accessibility level compared to WordPress 5.4.

**Comment by afercia** [comment]
Since the new buttons design applies to many other buttons in the UI and #24420 only changed the text color of some of these buttons, I don't think the accessibility and usability issue is fully solved and this should be further discussed and better addressed.

This is even more important considering that, sooner or later, the new Gutenberg design should be synced with the other admin pages. Extending the new buttons design to the rest of the admin would be a huge accessibility regression. 

Reopening.

**Comment by MichaelArestad** [comment]
> I don't think the accessibility and usability issue is fully solved and this should be further discussed and better addressed.

@afercia Can you please rename/rewrite this issue or create a new issue to discuss the broader topic as the specific topic of this issue was resolved: "Accessibility regression: "Save draft", "Switch to draft", and "Preview" buttons can't be visually identified as user interface controls"

I lean toward the creation of a new issue that clearly states the problem, goes over individual examples, and references this issue.

**Comment by afercia** [comment]
> @afercia Can you please rename/rewrite this issue or create a new issue to discuss the broader topic as the specific topic of this issue was resolved: "Accessibility regression: "Save draft", "Switch to draft", and "Preview" buttons can't be visually identified as user interface controls"

@MichaelArestad Id' like to point out that the specific issue initially reported is _not solved_. The only change to the "Save draft", "Switch to draft", and "Preview" buttons is that their color was changed from dark grey to blue.

<img width="1277" alt="Screenshot 2020-09-06 at 15 23 54" src="https://user-images.githubusercontent.com/1682452/92326849-68d69080-f055-11ea-8ed9-e4554fde8203.png">

Color alone is not sufficient 🙂 

I'll gladly rename this issue. As for the issue description, I think it has been explained sufficiently and it's clear that it applies to all the buttons with the new style, see for example https://github.com/WordPress/gutenberg/issues/23890#issuecomment-669990279

**Comment by ZebulanStanphill** [comment]
@afercia What do you think of this mockup?
![image](https://user-images.githubusercontent.com/19592990/94095719-edf8ce00-fde7-11ea-8c51-79c7d7fc3f12.png)
![image](https://user-images.githubusercontent.com/19592990/94095745-fc46ea00-fde7-11ea-9559-761d5f7c990a.png)

**Comment by afercia** [comment]
@ZebulanStanphill I think the look like beautiful buttons 🙂

**Comment by ZebulanStanphill** [comment]
More beautiful buttons: #25632.

**Comment by ZebulanStanphill** [comment]
Alright, we ran into some issues with the button design in #25632:

- The current design looks just like a text input. We need to come up with distinct visual differences (in shape, not merely color) to distinguish between all of the following:
  - plain text
  - links
  - text input
  - secondary button
  - primary button
- Concerns over removing button hierarchy. Do we actually need both an `isSecondary` and `isTertiary` style? The distinction seems rather arbitrary to me. And don't forget, `isSmall` and `isLink` (ugh) also exist.

My thinking is that all buttons should have a visible border unless they are in a toolbar. Of course, text inputs should also have borders, so how do we distinguish the two? Perhaps we can alter the borders of one or the other. I've seen some UIs use round borders for buttons and square borders for text inputs. How does that sound?

Simply underlining buttons is a bad idea since underlines are traditionally reserved for links. And remember, links have to use underlines (or something similar) because color alone is not enough to distinguish them from other text, due to colorblind users. So links are pretty locked in to their current style, as far as I can tell.

Also keep in mind that changing border thickness to indicate button vs. text input is also probably off the table due to that already being used to indicate unfocused versus focused... unless you want to redo focus styles.

And finally, remember that we need sufficient color contrast for whatever border/background colors we use.

So... any thoughts/ideas?

**Comment by afercia** [comment]
I do understand the concern raised by @enriquesanchez in #25632 

> At first glance, these buttons look like input fields to me. You can see how there's little to no visual difference between the proposed style for buttons and the 'Add tags' field:

> We need a balance that makes it clear these are buttons but that also not overwhelm a whole other set of users.

Re-posting here the screenshot used as example on #25632:

<img width="280" alt="Screen Shot 2020-09-28 at 11 04 20" src="https://user-images.githubusercontent.com/3276087/94457587-63fb8d00-017a-11eb-9eab-3f519d506479.png">

I'd like to note that in WordPress 5.4, before the buttons redesign, the shape of the buttons and the input fields was very similar: they were both "rectangles with rounded corners". What actually helped in clearly distinguish the buttons from the input fields was:
- the buttons had a default blue border
- the buttons had a default light grey background

Screenshot from WP 5.4:

<img width="340" alt="Screenshot 2020-09-30 at 14 17 16" src="https://user-images.githubusercontent.com/1682452/94684903-f93c8400-0328-11eb-8fc3-9e6b5b499895.png">

To me, this is just one more reason to revert the buttons style change 🙂 

Regarding the "tertiary" style: I never really understood why it was necessary. Historically, in WordPress there have been only two button styles: default and primary, where the latter should be exclusively used for the main action in a page.

**Comment by carolinan** [comment]
Since there is no progress reported on a new design for these buttons, can the change be reverted and the old style implemented while a new design is being worked on?

There doesn't seem to be any disagreement on this being a regression.

**Comment by cbirdsong** [comment]
Can this be revisited? I've had to explain that buttons look like inputs and vice versa to several clients in training sessions.

**Comment by afercia** [comment]
As of December 2024, this issue is still valid. The `default` and, more importantly, `tertiary` variants aren't accessible. These two variants need a new design.

In the screenshot below: default, primary, secondary, tertiary, link:

![Image](https://github.com/user-attachments/assets/dd457039-9e9c-4301-8069-6246c229baca)

**Comment by karmatosed** [comment]
Right now due to the amount of conversations here this feels like it needs to have feedback over yet more design. What might need to happen is discussion that leads to iterations, but having new design right now doesn't seem the path forward so let's focus the type of interactions this has through labels.
